lib/channels/index.ts

```diff
@@ -0,0 +1,43 @@
+import { AuthChannel, AuthChannelManagerOptions } from 'types';
+import { ALL_CHANNELS } from 'consts';
+import { LocalChannel } from './local';
+
+export class AuthChannelManager {
+  private options?: AuthChannelManagerOptions;
+
+  private activeChannels: AuthChannel[] = ['local'];
+
+  constructor(options?: AuthChannelManagerOptions) {
+    const channels = options?.channels;
+
+    if (typeof channels === 'string') {
```

Comment:
if you have to check the type like this it beats the whole purpose of typescript, it's ok to do it vanilla JS/ global scope variables